File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/C++/s221078931_commented.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

struct IntegralRect {
    int height;
    int width;
    IntegralRect(int h, int w) : height(h), width(w) {}
    
    int diagonal() const {
        if (height >= width) return INT_MAX;
        return height * height + width * width;
    }
    
    bool operator<(const IntegralRect& other) const {
        int d1 = diagonal(), d2 = other.diagonal();
        if (d1 != d2) return d1 < d2;
        return height < other.height;
    }
};

int main() {
    vector<IntegralRect> rects;
    for (int i = 0; i < 200; ++i) {
        for (int j = 0; j < 200; ++j) {
            rects.push_back(IntegralRect(i+1, j+1));
        }
    }
    sort(rects.begin(), rects.end());
    
    int h, w;
    while (cin >> h >> w) {
        if (h == 0 && w == 0) break;
        IntegralRect tmp(h, w);
        for (size_t i = 0; i < rects.size(); ++i) {
            if (tmp < rects[i]) {
                cout << rects[i].height << " " << rects[i].width << endl;
                break;
            }
        }
    }
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/comment/direct/Java/Go/s221078931_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

type IntegralRect struct {
	height int
	width  int
}

func (r IntegralRect) diagonal() int {
	if r.height >= r.width {
		return 1 << 60 // a very large number
	}
	return r.height*r.height + r.width*r.width
}

func main() {
	rects := make([]IntegralRect, 0, 200*200)
	for i := 0; i < 200; i++ {
		for j := 0; j < 200; j++ {
			h := i + 1
			w := j + 1
			rects = append(rects, IntegralRect{h, w})
		}
	}

	sort.Slice(rects, func(i, j int) bool {
		a := rects[i]
		b := rects[j]
		da := a.diagonal()
		db := b.diagonal()
		if da != db {
			return da < db
		}
		return a.height < b.height
	})

	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan() {
		line := scanner.Text()
		fields := strings.Fields(line)
		if len(fields) != 2 {
			continue
		}
		hStr, wStr := fields[0], fields[1]
		h, _ := strconv.Atoi(hStr)
		w, _ := strconv.Atoi(wStr)
		if h == 0 && w == 0 {
			break
		}
		tmp := IntegralRect{h, w}
		tmpDiag := tmp.diagonal()

		for i := 0; i < len(rects); i++ {
			r := rects[i]
			rd := r.diagonal()
			if rd > tmpDiag || (rd == tmpDiag && r.height > tmp.height) {
				fmt.Printf("%d %d\n", r.height, r.width)
				break
			}
		}
	}

	if err := scanner.Err(); err != nil {
		fmt.Fprintln(os.Stderr, "reading input:", err)
	}
}

// End of Code
